### PR TITLE
Casting, range check, and more compilation warning resolutions

### PR DESCRIPTION
**Pull request by bartwe**

### CODE REVIEW

File: src/gpu/d3d12/SDL_gpu_d3d12.c [hunk 1]
```diff
@@ -2673,13 +2674,21 @@ static D3D12Texture *D3D12_INTERNAL_CreateTexture(
 
     heapFlags = D3D12_HEAP_FLAG_NONE;
 
+    if (((textureCreateInfo->depth <= 1) && (!textureCreateInfo->isCube) && (textureCreateInfo->layerCount > UINT16_MAX))
+        || ((textureCreateInfo->depth > 1) && (textureCreateInfo->depth > UINT16_MAX))
+        || (textureCreateInfo->levelCount > UINT16_MAX)) {
+        SDL_LogError(SDL_LOG_CATEGORY_GPU, "Failed to create texture! Argument out of range.");
+        D3D12_INTERNAL_DestroyTexture(renderer, texture);
+        return NULL;
+    }
```
Comment (flibitijibibo) [inline src/gpu/d3d12/SDL_gpu_d3d12.c:2683]: This part looks good, can we put it in gpu.c possibly?

**Review by flibitijibibo — COMMENTED** [review]
Overall pretty safe, though I bet the create checks could apply to all backends.

File: src/gpu/d3d12/SDL_gpu_d3d12.c (hunk 1 above)
Comment (bartwe) [reply src/gpu/d3d12/SDL_gpu_d3d12.c:2683]: we could change levelCount/layerCount to UINT16 to avoid the possibility too.

**Comment by thatcosmonaut** [comment]
Most of this looks like it was written to satisfy MSVC at max warning level, which is not required upstream and therefore not necessary. Not really sure what's going on with the texture parameter validation but it should be moved into SDL_gpu.c and rewritten to follow the pattern of our other validations there anyway.

**Pull request closed without merging**